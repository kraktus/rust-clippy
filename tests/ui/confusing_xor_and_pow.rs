#![allow(unused)]
#![warn(clippy::confusing_xor_and_pow)]
#![allow(clippy::eq_op)]

fn main() {
    // Should warn:
    let _ = 2i32 ^ 9i32;
    let _ = 50i32 ^ 3i32;
    let _ = 5i32 ^ 8i32;
    let _ = 2i32 ^ 32i32;

    // Should not warn:
    let _ = 0x02 ^ 5;
    let _ = 2 ^ 0x02;
    let _ = 10 ^ 0b0101;
}
