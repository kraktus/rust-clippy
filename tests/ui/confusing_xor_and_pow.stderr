error: '^' is not the exponentiation operator
  --> $DIR/confusing_xor_and_pow.rs:7:13
   |
LL |     let _ = 2i32 ^ 9i32;
   |             ^^^^^^^^^^^ help: did you mean to write: `2i32.pow(9)`
   |
   = note: `-D clippy::confusing-xor-and-pow` implied by `-D warnings`

error: '^' is not the exponentiation operator
  --> $DIR/confusing_xor_and_pow.rs:8:13
   |
LL |     let _ = 50i32 ^ 3i32;
   |             ^^^^^^^^^^^^ help: did you mean to write: `50i32.pow(3)`

error: '^' is not the exponentiation operator
  --> $DIR/confusing_xor_and_pow.rs:9:13
   |
LL |     let _ = 5i32 ^ 8i32;
   |             ^^^^^^^^^^^ help: did you mean to write: `5i32.pow(8)`

error: it appears that you are trying to get the maximum value of an integer, but '^' is not exponentiation operator
  --> $DIR/confusing_xor_and_pow.rs:10:13
   |
LL |     let _ = 2i32 ^ 32i32;
   |             ^^^^^^^^^^^^ help: try with: `u32::MAX`

error: aborting due to 4 previous errors

